apps/dapp/src/providers/SpiceBazaarProvider.tsx

```diff
@@ -745,6 +749,70 @@ export const SpiceBazaarProvider = ({ children }: PropsWithChildren) => {
     }
   }, [wallet, signer, papi]);
 
+  const getDelegatedAddress = useCallback(async () => {
+    if (!wallet) {
+      return '';
+    }
+
+    try {
+      const templeGoldStakingContract = (await papi.getContract(
+        getAppConfig().contracts.templeGoldStaking
+      )) as TempleGoldStaking;
+
+      const delegatedAddress = await templeGoldStakingContract.delegates(
+        wallet
+      );
+
+      // Return empty string if no delegate (returns 0x0000000000000000000000000000000000000000)
+      return delegatedAddress === '0x0000000000000000000000000000000000000000'
```

Comment:
I have seen the zero address reused a couple of times. Would it help to have it as a global constant?

Reply:
Good catch, actually we have this already, will use it.

Reply:
https://github.com/TempleDAO/temple/pull/1197/commits/f2de04c1b629f5bd801bcade6a7abcf486890883